submariner/cable-driver-protocols-enhancement.md

```diff
@@ -0,0 +1,179 @@
+# Cable Driver Protocol Support Enhancements
+
+## Summary
+
+Submariner today supports three cable drivers: VXLAN, IPSec (tunnel mode) and WireGuard. The
+goal of this proposal is to outline additional protocols and enhancements for the Submariner
+Cable Driver. These include the additional support of IP-in-IP for unencrypted connections
+between clusters and the support of IPSec transport mode.
+
+For IP in IP, the benefits fall into the following categories:
```

Comment:
I'm not familiar with cloud-prepare bits, I assume we need to run some specific firewall config to allow IPinIP traffic, right?

Reply:
So it depends on the cloud env - for GCP and for AWS it looks like some simple configuration... For Azure IP in IP traffic is blocked... 

It looks like it's also supported in OpenShift but in concert with the edge loadbalancer from F5... But I'm guessing from a configuration POV it shouldn't pose a problem

The goal here though is to enable a low overhead encapsulation that can be used together with IPSec in transport mode - so the IPinIP traffic would be encaped with the ESP header[Outer IP | ESP Header | Payload (inner IP | UDP or TCP | inner payload )] and this should not pose any additional issues for various environments as it's transparent till arrival at the GW...

Reply:
Ack,
I think it makes sense to add to the proposal section that describes the required changes in cloud-prepare to support IPinIP traffic.